Update Anthropic tests for error responses and parallelism

kpavlov opened the pull request:
<!--
Thank you so much for your contribution!

Please fill in all the sections below.
Please open the PR as a draft initially. Once it is reviewed and approved, we will ask you to add documentation and examples.
Please note that PRs with breaking changes or without tests will be rejected.

Please note that PRs will be reviewed based on the priority of the issues they address.
We ask for your patience. We are doing our best to review your PR as quickly as possible.
Please refrain from pinging and asking when it will be reviewed. Thank you for understanding!
-->

## Issue
<!-- Please specify the ID of the issue this PR is addressing. For example: "Closes #1234" or "Fixes #1234" -->
Improve tests for Anthropic Chat Language Models

## Change

This PR introduces improved testing for the `AnthropicChatModel`. 

- Added unit tests for error handling across various [HTTP status codes](https://docs.anthropic.com/en/api/errors), ensuring proper exceptions and messages. 
- Enhancements also enable JUnit parallel test execution for better efficiency and restrict test runs to environments with appropriate API keys. 
- [ai-mocks-anthropic](https://kpavlov.github.io/ai-mocks/docs/ai-mocks-anthropic/) dependency has been added to aid in testing
- Refine system prompt in the Anthropic integration test for German translation agent to return only translation and nothing else
- Reformat code in affected files


## General checklist
<!-- Please double-check the following points and mark them like this: [X] -->
- [x] There are no breaking changes
- [x] I have added unit and/or integration tests for my change
- [x] The tests cover both positive and negative cases
- [ ] I have manually run all the unit and integration tests in the module I have added/changed, and they are all green
- [ ] I have manually run all the unit and integration tests in the [core](https://github.com/langchain4j/langchain4j/tree/main/langchain4j-core) and [main](https://github.com/langchain4j/langchain4j/tree/main/langchain4j) modules, and they are all green
<!-- Before adding documentation and example(s) (below), please wait until the PR is reviewed and approved. -->
- [ ] I have added/updated the [documentation](https://github.com/langchain4j/langchain4j/tree/main/docs/docs)
- [ ] I have added an example in the [examples repo](https://github.com/langchain4j/langchain4j-examples) (only for "big" features)
- [ ] I have added/updated [Spring Boot starter(s)](https://github.com/langchain4j/langchain4j-spring) (if applicable)


## Checklist for adding new maven module
<!-- Please double-check the following points and mark them like this: [X] -->
- [ ] I have added my new module in the root `pom.xml` and `langchain4j-bom/pom.xml`


## Checklist for adding new embedding store integration
<!-- Please double-check the following points and mark them like this: [X] -->
- [ ] I have added a `{NameOfIntegration}EmbeddingStoreIT` that extends from either `EmbeddingStoreIT` or `EmbeddingStoreWithFilteringIT`
- [ ] I have added a `{NameOfIntegration}EmbeddingStoreRemovalIT` that extends from `EmbeddingStoreWithRemovalIT`

## Checklist for changing existing embedding store integration
<!-- Please double-check the following points and mark them like this: [X] -->
- [ ] I have manually verified that the `{NameOfIntegration}EmbeddingStore` works correctly with the data persisted using the latest released version of LangChain4j

kpavlov committed:
Add AI Mocks dependency for testing in langchain4j-anthropic

Updated `langchain4j-parent` and `langchain4j-anthropic` POMs to include `ai-mocks` (version 0.1.9) as a test dependency.

kpavlov committed:
Enable tests based on environment variables for API keys

Added `@EnabledIfEnvironmentVariable` annotations to several test classes and methods to ensure they run only when the required API keys, such as `ANTHROPIC_API_KEY`, are present. This change prevents unnecessary test execution in environments lacking the appropriate configurations, improving testing robustness and flexibility.

kpavlov committed:
Enable JUnit parallel test execution with dynamic strategy

Add configuration to enable parallel test execution in JUnit. This allows tests and classes to run concurrently, improving test efficiency. The dynamic strategy is used to optimize resource utilization during execution.

kpavlov committed:
Add unit tests for AnthropicChatModel error handling

Introduce tests to validate proper error handling for various HTTP status codes in AnthropicChatModel. These tests ensure the model correctly throws exceptions and provides expected messages for different error scenarios such as authentication errors, rate limits, and server issues.

kpavlov committed:
Revert OpenAiChatModelListenerIT

kpavlov committed:
reformat code

kpavlov committed:
better system prompt for translation test

kpavlov committed:
Merge branch 'main' into anthropic-tests

# Conflicts:
#	langchain4j-anthropic/src/test/java/dev/langchain4j/model/anthropic/AnthropicChatModelIT.java
#	langchain4j-anthropic/src/test/java/dev/langchain4j/model/anthropic/AnthropicStreamingChatModelIT.java
#	langchain4j-core/src/test/java/dev/langchain4j/model/chat/ChatModelListenerIT.java
#	langchain4j-core/src/test/java/dev/langchain4j/model/chat/StreamingChatModelListenerIT.java

dliubarskyi commented:
@kpavlov thanks a lot, that looks great!

kpavlov committed:
merged with main, reformat code

dliubarskyi commented on langchain4j-core/src/test/java/dev/langchain4j/model/chat/ChatModelListenerIT.java:
```diff
@@ -78,6 +78,8 @@ protected Integer maxTokens() {
     protected abstract Class<? extends Exception> expectedExceptionClass();
 
     @Test
+    @EnabledIfEnvironmentVariable(named = "ANTHROPIC_API_KEY", matches = ".+")
```
This test is not only for anthropic, but for other providers as well

kpavlov replied:
ahh! sorry! indeed!

dliubarskyi commented on langchain4j-core/src/test/java/dev/langchain4j/model/chat/StreamingChatModelListenerIT.java:
```diff
@@ -78,6 +78,7 @@ protected Integer maxTokens() {
     protected abstract Class<? extends Exception> expectedExceptionClass();
 
     @Test
+    @EnabledIfEnvironmentVariable(named = "ANTHROPIC_API_KEY", matches = ".+")
```
same here

dliubarskyi commented on langchain4j/src/test/java/dev/langchain4j/service/common/AbstractAiServiceIT.java:
```diff
@@ -41,14 +41,14 @@ interface Assistant {
 
     @ParameterizedTest
     @MethodSource("models")
+    @EnabledIfEnvironmentVariable(named = "ANTHROPIC_API_KEY", matches = ".+")
```
same here

dliubarskyi commented on langchain4j/src/test/java/dev/langchain4j/service/common/AbstractAiServiceIT.java:
```diff
@@ -82,6 +85,7 @@ void should_answer_simple_question(ChatLanguageModel model) {
     @ParameterizedTest
     @MethodSource("modelsSupportingToolsAndJsonResponseFormatWithSchema")
     @EnabledIf("supportsToolsAndJsonResponseFormatWithSchema")
+    @EnabledIfEnvironmentVariable(named = "ANTHROPIC_API_KEY", matches = ".+")
```
same here

dliubarskyi commented on langchain4j/src/test/java/dev/langchain4j/service/common/AbstractStreamingAiServiceIT.java:
```diff
@@ -40,6 +40,7 @@ interface Assistant {
 
     @ParameterizedTest
     @MethodSource("models")
+    @EnabledIfEnvironmentVariable(named = "ANTHROPIC_API_KEY", matches = ".+")
```
same here

kpavlov committed:
revert common integration tests

kpavlov committed:
Enable tests conditionally based on `ANTHROPIC_API_KEY` environment variable.

This update ensures that integration tests for Anthropic AI services only run when the `ANTHROPIC_API_KEY` environment variable is set. This prevents test failures in environments where the required API key is unavailable.

dliubarskyi reviewed (approved):
@kpavlov great job, thank you!
